modMcf/test/org/aion/mcf/account/AccountManagerTest.java

```diff
@@ -0,0 +1,320 @@
+    }
+
+    @Test
+    public void testGetKeyNotInMap(){
+        // check that there are currently no accounts in the manager
+        assertThat(accountManager.getAccounts().size()).isEqualTo(0);
+
+        // try to get a key not in the manager
+        assertNull(accountManager.getKey(Address.wrap(k1.getAddress())));
+    }
+
+    @Test
+    public void testUnlockAndLockMultipleTimes(){
+        // first an account
+        assertTrue(accountManager.unlockAccount(Address.wrap(k1.getAddress()), p1, AccountManager.UNLOCK_DEFAULT));
+        assertThat(accountManager.getAccounts().size()).isEqualTo(1);
+
+        // lock k1 and check that timeout is changed
+        assertTrue(accountManager.lockAccount(Address.wrap(k1.getAddress()), p1));
+        List<Account> accountsList;
+        accountsList = accountManager.getAccounts();
+        assertThat(accountsList.size()).isEqualTo(1);
+        assertThat(accountsList.get(0).getTimeout()).isLessThan(Instant.now().getEpochSecond());
+
+        // now unlock account with k1 again and check that timeout is changed
+        assertTrue(accountManager.unlockAccount(Address.wrap(k1.getAddress()), p1, AccountManager.UNLOCK_DEFAULT));
+        assertThat(accountManager.getAccounts().size()).isEqualTo(1);
+        assertThat(accountsList.get(0).getTimeout()).isEqualTo(Instant.now().getEpochSecond() + AccountManager.UNLOCK_DEFAULT);
+
+    }
+
+    private static void cleanAccountManager(){
+        // lock all the accounts, which modifies the timeout
+        accountManager.lockAccount(Address.wrap(k1.getAddress()), p1);
+        accountManager.lockAccount(Address.wrap(k2.getAddress()), p2);
+        accountManager.lockAccount(Address.wrap(k3.getAddress()), p3);
+
+        // remove accounts
+        accountManager.getKey(Address.wrap(k1.getAddress()));
+        accountManager.getKey(Address.wrap(k2.getAddress()));
+        accountManager.getKey(Address.wrap(k3.getAddress()));
+
+        // check that manager is cleared
+        assertThat(accountManager.getAccounts().size()).isEqualTo(0);
+    }
+
+    private static void cleanFiles(){
+        // get a list of all the files in keystore directory
+        File folder = new File(KEYSTORE_PATH);
+        File[] AllFilesInDirectory = folder.listFiles();
+        List<String> allFileNames = new ArrayList<>();
+        List<String> filesToBeDeleted = new ArrayList<>();
+
+        // check for invalid or wrong path - should not happen
+        if(AllFilesInDirectory == null)
+            return;
+
+        for(File file: AllFilesInDirectory){
+            allFileNames.add(file.getName());
+        }
+
+        // get a list of the files needed to be deleted, check the ending of file names with corresponding addresses
+        for(String name: allFileNames){
+            String ending = name.substring(name.length()-64);
+
+            if(ending.equals(address1) || ending.equals(address2) || ending.equals(address3)) {
```

Comment:
Thanks for taking the time to ensure you're only deleting files actually created by your test